src/jsx/absrel.jsx

```diff
@@ -155,35 +155,56 @@ var BSRELSummary = React.createClass({
 var BSREL = React.createClass({
   float_format: d3.format(".2f"),
 
-  loadFromServer: function() {
+  processData: function(data){
+    var test_results = _.mapObject(data['branch attributes']['0'], (val, key) => {
+      var tested = val.LRT != null;
 
-    var self = this;
+      return {
+        LRT: tested ? val.LRT : 'test not run',
+        p: tested ? val['Corrected P-value'] : 1,
+        'uncorrected p': tested ? val['Uncorrected P-value'] : 1,
+        tested: tested
+      };
+    });
 
-    d3.json(this.props.url, function(data) {
-      data["fits"]["MG94"]["branch-annotations"] = self.formatBranchAnnotations(
-        data,
-        "MG94"
-      );
-      data["fits"]["Full model"][
-        "branch-annotations"
-      ] = self.formatBranchAnnotations(data, "Full model");
-
-      // GH-#18 Add omega annotation tag
-      data["fits"]["MG94"]["annotation-tag"] = "ω";
-      data["fits"]["Full model"]["annotation-tag"] = "ω";
-
-      self.setState({
-        annotations: data["fits"]["Full model"]["branch-annotations"],
-        json: data,
-        pmid: data["PMID"],
-        fits: data["fits"],
-        full_model: data["fits"]["Full model"],
-        test_results: data["test results"],
-        input_data: data["input_data"],
-        tree: d3.layout.phylotree()(data["fits"]["Full model"]["tree string"])
-      });
+    data["fits"]["Full adaptive model"][
+      "branch-annotations"
+    ] = this.formatBranchAnnotations(data, "Full adaptive model");
+
+    data["fits"]["Baseline MG94xREV"]["branch-annotations"] = this.formatBranchAnnotations(
+      data,
+      "Baseline MG94xREV"
+    );
+
+    // GH-#18 Add omega annotation tag
+    data["fits"]["Baseline MG94xREV"]["annotation-tag"] = "ω";
+    data["fits"]["Full adaptive model"]["annotation-tag"] = "ω";
+
+    data["trees"] = {
+      branchLengths: {
+        "Baseline MG94xREV": _.mapObject(data["branch attributes"][0], val=>val["Baseline MG94xREV"]),
+        "Full adaptive model": _.mapObject(data["branch attributes"][0], val=>val["Full adaptive model"])
+      }
+    }
+    _.each(_.keys(data.fits), model=>{delete data.fits[model]['Rate Distributions'];});
+    this.setState({
+      annotations: data["fits"]["Full adaptive model"]["branch-annotations"],
+      json: data,
+      pmid: data["PMID"],
+      fits: data["fits"],
+      full_model: data["fits"]["Full adaptive model"],
+      test_results: test_results,
+      input_data: data["input"],
+      tree: d3.layout.phylotree()(data.input['trees'][0]),
+      branch_attributes: data["branch attributes"][0]
     });
+  },
 
+  loadFromServer: function() {
```

Comment:
At some point in the future, perhaps when migrating to es6, we can remove loadFromServer functions and place that in separate component. - per @stephenshank recommendation